### PR TITLE
updating size to less than occupied size should fail

### PR DESCRIPTION
**Pull request by Kishan-Dhakan**
- Adds coverage to update allocation.

Related PR: https://github.com/0chain/0chain/pull/1113

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/alloc-update  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | zero-lock-validation  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/alloc-update  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | zero-lock-validation  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by peterlimg — APPROVED** [review]
lgtm

File: tests/cli_tests/zboxcli_update_allocation_test.go [hunk 1]
```diff
@@ -180,6 +180,42 @@ func TestUpdateAllocation(t *testing.T) {
...
+
+		output, err := uploadFile(t, configPath, map[string]interface{}{
+			"allocation": allocationID,
+			"remotepath": "/dir/",
+			"localpath":  filename,
+		}, true)
+		require.Nil(t, err, strings.Join(output, "\n"))
+		require.Len(t, output, 2)
+
+		size := int64(-9000) // reducing it by 9000 should fail since 2048 is being used
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_update_allocation_test.go:200]: nice test case to have

**Pull request merged**